src/mdacli/save.py

```diff
@@ -40,108 +41,65 @@ def save_results(fprefix, results):
-                # of entries
-                splitted_item = np.split(
-                    item,
-                    item.shape[min_dim],
-                    axis=min_dim,
-                    )
-
-                for arr in splitted_item:
-                    files_to_zip.append(f"{key}_dim_{min_dim}_idx_{i}.csv")
-                    np.savetxt(
-                        fname=files_to_zip[-1],
-                        X=np.squeeze(arr),
-                        delimiter=',',
-                        )
-
-                # Compress all csv files into a single zip archive
-                with zipfile.ZipFile(f'{key}.zip', 'w') as zipF:
-                    for file_name in files_to_zip:
-                        zipF.write(
-                            file_name,
-                            compress_type=zipfile.ZIP_DEFLATED,
-                            )
-                        os.remove(file_name)
-
-            elif n_dims > 3:
-                np.save(
-                    f"{fprefix}_{key}",
-                    item,
-                    allow_pickle=False,
-                    )
-
-            else:
-                warnings.warn(
-                    Emphasise.warning(
-                        "Saving numpy arrays with more than "
-                        "three dimensions is currently not supported."
-                        )
-                    )
-
-        elif (isinstance(item, (bool, int, float, list, tuple, dict))
-              or item is None):
-            # This can be encoded in a json file
-            json_dict[key] = item
-
-        else:
-            warnings.warn(Emphasise.warning(f"Saving {key} of type {type(item)}"
-                          "is currently not supported."))
-
-    # Stack 1D arrays and save teheem to csv
-    if len(list_1D) > 0:
-        out_lists, out_lables = stack_1d_arrays_list(list_1D, list_1D_labels)
-
-        command_run = ' '.join(sys.argv) + "\n"
-
-        for out_list, out_label in zip(out_lists, out_lables):
-            out_label = out_label.flatten()
-
-            # [3:] to align lables with entries
-            savetxt_w_command(
-                fname=f"{fprefix}_{'_'.join(out_label)}.csv",
-                X=out_list.T,
-                header=''.join([f"{i:>25}" for i in out_label])[3:]
-                )
 
-    # Save everything which is left to a json file
-    # if json dict is not empty
-    if json_dict:
-        with open(f'{fprefix}.json', 'w') as f:
-            json.dump(json_dict, f)
+    for key, value in results.items():
+        if is_array_1D(value):
+            list_1D.append(value)
+            list_1D_labels.append(key)
 
-    return
+    return list_1D, list_1D_labels
+
+
+def is_array_1D(arr):
+    """Assert arr is an array and of 1 single dimension."""
+    valid = \
+        isinstance(arr, np.ndarray) \
+        and arr.ndim == 1
+
+    return valid
 
 
 def stack_1d_arrays_list(list_1D, extra_list=None):
```

Comment:
One thing I would like to do when I wrote this function is not to restrict this to one list and an extra list but apply the operations on `list_1D` to a series of lists. The function definition could be something like

```python
def stack_1d_arrays_list(list_1D, *args):
"""Stack a list of 1D numpy arrays of the same length vertically together.    

The result is a list containing 2D arrays where each array got the same    
number of rows. Yo can apply additional lists of 1d arrays as extra arguments. 
The operations applied to the first are also applied to the 
consecutive lists of arrays.

Parameters    
----------    
list_1d : list
   list of 1 dimensional numpy arrays
```

Reply:
I haven't changed that function. I am not sure yet about what you mean. I will look to it carefully later and "discord" with you. Anyway, we are meeting today at 11PM, right?

Reply:
Yes lets discuss this on discord. I can also take a look at this later.

Reply:
I think we should address this on the `save` branch